ui/perfherder/CompareSubtestDistributionView.jsx

```diff
@@ -0,0 +1,376 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCog } from '@fortawesome/free-solid-svg-icons';
+import { react2angular } from 'react2angular/index.es2015';
+
+import perf from '../js/perf';
+import { errorMessageClass } from '../helpers/constants';
+import ErrorBoundary from '../shared/ErrorBoundary';
+import Graph from '../shared/Graph';
+import RepositoryModel from '../models/repository';
+import PerfSeriesModel from '../models/perfSeries';
+import PushModel from '../models/push';
+import { getData } from '../helpers/http';
+import { createApiUrl, perfSummaryEndpoint } from '../helpers/url';
+
+import RevisionInformation from './RevisionInformation';
+
+// TODO remove $stateParams after switching to react router
+export default class CompareSubtestDistributionView extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dataLoading: this.props.dataLoading,
```

Comment:
I would let this component be completely responsible for the `dataLoading` state - rather than using props - since it is fetching its own data. It can either have a default of true or setState to true in `componentDidMount`.